File: 165. Compare Version Numbers/Python/Compare Version Numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        tokens_1 = version1.split('.')
        tokens_2 = version2.split('.')
        length_1 = len(tokens_1)
        length_2 = len(tokens_2)
        for i in range(max(length_1, length_2)):
            in_int_1 = int(tokens_1[i]) if i < length_1 else 0
            in_int_2 = int(tokens_2[i]) if i < length_2 else 0
            if in_int_1 > in_int_2:
                return 1
            elif in_int_1 < in_int_2:
                return -1
        return 0

File: 165. Compare Version Numbers/Java/Compare Version Numbers.java
public class Solution {
    public int compareVersion(String version1, String version2) {
        String[] tokens1 = version1.split("\\.");
        String[] tokens2 = version2.split("\\.");
        int length1 = tokens1.length;
        int length2 = tokens2.length;
        for (int i = 0; i < Math.max(length1, length2); i++) {
            int inInt1 = i < length1 ? Integer.parseInt(tokens1[i]) : 0;
            int inInt2 = i < length2 ? Integer.parseInt(tokens2[i]) : 0;
            if (inInt1 > inInt2) {
                return 1;
            } else if (inInt1 < inInt2) {
                return -1;
            }
        }
        return 0;
    }
}